### PR TITLE
fix: fix clocked schedule

### PR DESCRIPTION
**Pull request by yywing**
Clocked schedule is stateless.
It can run once because of PeriodicTask.one_off.

### CODE REVIEW

File: django_celery_beat/models.py [hunk 1]
```diff
@@ -186,12 +186,6 @@ class ClockedSchedule(models.Model):
         verbose_name=_('Clock Time'),
         help_text=_('Run the task at clocked time'),
     )
-    enabled = models.BooleanField(
```
Comment (auvipy) [inline django_celery_beat/models.py:189]: this is a breaking change

Comment (yywing) [reply django_celery_beat/models.py:189]: > can you explain more of your reason for this approach?

I maybe handle it before this weekends(timezone: Beijing). too busy now.

Comment (auvipy) [reply django_celery_beat/models.py:189]: we can introduce swapalbe/ abstract base model as Django-activity-stream did

**Comment by yywing** [comment]
I will work on this PR.
And I maybe pull a new feature for mult time schedule.

**Comment by yywing** [comment]
图片

**Comment by yywing** [comment]
1. `is _due` return `None` will make schedule stuck. I dont know why and I canot find answer on celery document. But it works.
2. Clocked schedule changed clocked schedule model, and it will make schedule reload. As picture I upload.  
    1) I create many clocked schedule which work on the same time.
    2) first clocked schedule run ，and change model，and schedule reload.
    3) reload wiled call `is_due` and all clocked schedule set to false, but it olny called one
    4) schedule change and all schedule reload, all scheled set false at 2),so never called.

So only two task run.

**Comment by yywing** [comment]
I change clocked model, remove enabled field, use `PeriodicTask.one_off`.

File: django_celery_beat/models.py (hunk 1 above)
Comment (yywing) [reply django_celery_beat/models.py:204]: Or we can keep it but it always True.

**Comment by superandrew** [comment]
I am looking forward for this PR, @yywing it is great that you have been able to track, understand fix one of the most hideous and awkward bugs of this extension.

**Comment by superandrew** [comment]
just to add some hype: pip install git+https://github.com/yywing/django-celery-beat/@f2bdc5214b3c70ee3805654855ab08e55452df69
fixes the problem reported on #310

**Comment by superandrew** [comment]
hi, any news on when this PR is going to be released?

**Comment by auvipy** [comment]
there are some concerns regarding backward compat

**Comment by superandrew** [comment]
too bad. My experience with my use case and django_celery_beat was that everything worked for days before stop working randomly (then I understood it was not literally randomly but there was a reason), and this fix definitely fixed everything. So I really hope this could go in production as soon as possible. I am not confident with celery or this package, but if I can help with some real world testing I'd be more than happy to do so.

**Comment by auvipy** [comment]
please do some more real-world testing and report back

File: django_celery_beat/models.py (hunk 1 above)
Comment (auvipy) [reply django_celery_beat/models.py:204]: what about default=False?

**Comment by superandrew** [comment]
my production environment started using this package about 10 days ago.

We were able to run hundreds of task, and the scheduled tasks never stopped once.

Before this fix, after a few days, some condition would make celery beat block on one task and stop process the tasks all together.

I hope this helps. Please let me know if there is anything specific I could test.

File: django_celery_beat/models.py (hunk 1 above)
Comment (yywing) [reply django_celery_beat/models.py:204]: But it act as always True.

**Comment by sfdye** [comment]
Is it possible to release this fix please?

**Comment by auvipy** [comment]
Next week Hopefully

**Comment by adl-asi** [comment]
@auvipy Will this be merged soon? Trying to use clocked schedule in a production setting and running into issues

**Comment by adl-asi** [comment]
Just checking in again. Is there an eta for when this fix will be released?

**Comment by auvipy** [comment]
hope full this week. ping me again

**Comment by saethor** [comment]
Any updates about this?

**Comment by belongwqz** [comment]
still waiting

**Comment by Lotram** [comment]
@auvipy  any news?

**Comment by revoteon** [comment]
@auvipy Any updates?

**Comment by ghazi-git** [comment]
I really wish I'd seen this PR earlier. I started using clocked schedule last month. Like everyone else, things started working fine and then at some point, the clocked tasks stopped being processed. What I found out is that whenever there is one task where `PeriodicTask.enabled` is `True` but `ClockedSchedule.enabled` is `False`, all tasks after it are never executed. So, the question became how did I end up in a situation where the task `PeriodicTask.enabled` is `True` but `ClockedSchedule.enabled` is `False`.

Later I was able to reproduce it consistently by scheduling a task, then stopping celery beat until the scheduled time of the task has passed. When celery beat is brought back up after that, the scheduled task, that should have executed when celery beat was down, is picked up and gets executed. So, `ClockedSchedule.enabled` is set to `False`, but then `PeriodicTask.enabled` is never changed to `False`, which results in the next tasks not executed. In production, that might be translated to "a clocked task needs to be scheduled to execute when there is some downtime/maintenance for the issue to start happening"

Anyway, right before deciding to switch to one-off tasks using crontab schedule, I saw that there is a new version and when checking the commits , I found this PR. Now, that `ClockedSchedule.enabled` is removed, I installed the new version and retested the issue and it is no longer happening. So, thank you @yywing for fixing this and @auvipy for releasing the fix.

**Comment by sfdye** [comment]
Looks like `2.1.0` was released yesterday
https://pypi.org/project/django-celery-beat/#history

**Pull request merged**